markdown: remove unnecessary blockquote `>`

shisama opened the pull request:
<!-- Please provide a brief summary of your changes: -->
Fix #9099

This removes unnecessary blockquote `>` when the paragraph is made a single line.
I am not confident my implementation is good. Please tell me if you know a better way to fix this.

<!-- Please ensure you’ve done all of these things (if applicable). -->
<!-- You can replace the `[ ]` with `[x]` to mark each task as done. -->

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

thorn0 commented:
This solution seems to be fighting the consequences, not the reason of the bug. `>` shouldn't end up in a node of the type `word` in the first place.

kachkaev commented:
Once the right solution is found, it'd be also great to add ` changelog_unreleased/markdown/pr-9125.md` file, similar to a markdown in #9078

thorn0 commented:
Now this breaks `--prose-wrap preserve`. Also please make sure to test nested blockquotes.

thorn0 commented:
A solution that involves replacing `>` in a string can't be correct. If `>` has been already stringified, it's too late to meddle. It shouldn't be there. Why does it get there?

shisama commented:
@thorn0 Thanks. I found the cause that `>` is included. `options.originalText.slice()` in the below code transforms `\n` `\n> `. So I fixed the condition of this. But I'm not confident if the solution is correct.
Please take a look. https://github.com/prettier/prettier/pull/9125/commits/d43f64d6832bb0399171395c4ab512b8b24aa206

And I added a test for nested blockquotes. https://github.com/prettier/prettier/pull/9125/commits/64f13fcd8520e08cbc075e47e23b554c0bfc04d6

https://github.com/prettier/prettier/blob/810325daf685173f5f43e7ad1ac7603c4687a6d1/src/language-markdown/preprocess.js#L47-L57

sosukesuzuki commented on changelog_unreleased/markdown/pr-9125.md:
```diff
@@ -0,0 +1,15 @@
+> `x` > `y`
+
+// Prettier master
+> `x` `y`
```
This output is incorrect with our default options(`--prose-wrap=preserve `).
Can you change this to
```md
> `x`
> `y`
```
?

***

**Prettier pr-9125**
[Playground link](https://deploy-preview-9125--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEA+ABAAwB6YDpQaYCe+UIANCBAA4wCW0AzsqAIYBOHEA7gAqcELFGwA2PNsRZUARhzZgA1nBgBlGgvpQA5shgcArnCpwAtjLgATS1YAybHQbba4AMQgdTbGAx3IQbAYwEJQgABYwpqIA6mH08EwaYHCqQvH0AG7xxP5gTNIgWkxwHDB88tpeyABmYsVUAFZM2ABC8koqqmymcLZacDV1xiBN2Kpa2qJwAIoGEPCDovUgGhzFHP5eHIqWvORUNBxaMNH0ljBhyAAcAAwH3MXR8jT+h3DrGQNUAI5z8OW0YQBJgAWigcCsVlCHDgv3oMPKziqSFqS2GxVM9D0hnREyms3mAxRQyoMDYMlO50uSAATKT5PRRBMAMIQUzIkDvACsoQMxQAKuThKjlhkjABJKA2WCqMBHOgAQSlqhgxCmi2KAF9NUA)
```sh
--parser markdown
```

**Input:**
```markdown
> `x`
> `y`

```

**Output:**
```markdown
> `x`
> `y`

```

sosukesuzuki commented on tests/markdown/blockquote/paragraph.md:
```diff
@@ -1,2 +1,22 @@
 > This is a long long long long long long long long long long long long long long long paragraph.
 > This is a long long long long long long long long long long long long long long long paragraph.
+
```
Can you add tests for `prose-wrap=never` and `prose-wrap=preserve`?

sosukesuzuki reviewed (commented):
Can you add tests for `prose-wrap=never` and `prose-wrap=preserve` with updating `tests/markdown/jsfmt.spec.js`?

kachkaev commented:
👋 @shisama, would you be interested in updating this PR to help us close #9099?
